### PR TITLE
verify and materialize resource backrefs

### PR DESCRIPTION
**Pull request by geoberle**
data file fields that are tagged to be references to resources by `/common-1.json#/definitions/resourceref` get their reference verified. those data files are also back referenced from the resources themselves. the bundle representation of a reference will look like this

```json
{
    ...
    "resources": {
        "/path/to/resource": {
            "content": "...",
            ...
            "backrefs": [
                {
                    "path": "/path/to/datafile.yml",
                    "datafileSchema": "/openshift/namespace-1.yml",
                    "type": "Namespace_v1",
                    "jsonpath": "jsonpath to element that references a resource"
                },
            ]
        }
    }
    ...
}
```

Depends on https://github.com/app-sre/qontract-schemas/pull/160

References:

* design doc https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/app-sre/design-docs/qontract-schema-resource-references.md
* https://issues.redhat.com/browse/APPSRE-5629

Signed-off-by: Gerd Oberlechner <goberlec@redhat.com>

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -9,3 +9,6 @@ deps =
 [testenv:flake8]
 commands = flake8 validator
 deps = flake8~=3.8
+
+[flake8]
+max-line-length = 120
```
Comment (fishi0x01) [inline tox.ini:14]: In qontract-reconcile we agreed on `88`. Should we keep this uniform across our repos?

File: validator/bundle.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        return {
+            "git_commit": self.git_commit,
+            "git_commit_timestamp": self.git_commit_timestamp,
+            "schemas": self.schemas,
+            "graphql": self.graphql,
+            "data": self.data,
+            "resources": self.resources
+        }
+
+    def get_graphql_type_for_schema(self, schema) -> Optional[GraphqlType]:
```
Comment (fishi0x01) [inline validator/bundle.py:67]:
```
schema: str
```

File: tox.ini (hunk 1 above)
Comment (geoberle) [reply tox.ini:14]: i was somehow under the impression we were at 120. i will revert to 88.

File: validator/bundle.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        return self._top_level_schemas.get(schema)
+
+    def get_graphql_type_by_name(self, type: str) -> Optional[GraphqlType]:
+        return self._graphql_type_by_name.get(type)
+
+    def is_top_level_schema(self, datafile_schema) -> bool:
+        return datafile_schema in self._top_level_schemas
+
+
+def load_bundle(bundle_source) -> Bundle:
```
[the remark this answers is not in the document]
Comment (geoberle) [reply validator/bundle.py:77]: *picard facepalm* thx!

File: validator/bundle.py (hunk 2 above)
Comment (geoberle) [reply validator/bundle.py]: [f87fece](https://github.com/app-sre/qontract-validator/pull/37/commits/f87fecebc3cf717084bc423466e3ebc4e691e399)

File: validator/bundle.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (geoberle) [reply validator/bundle.py]: [f87fece](https://github.com/app-sre/qontract-validator/pull/37/commits/f87fecebc3cf717084bc423466e3ebc4e691e399)

File: tox.ini (hunk 1 above)
Comment (geoberle) [reply tox.ini]: [f87fece](https://github.com/app-sre/qontract-validator/pull/37/commits/f87fecebc3cf717084bc423466e3ebc4e691e399)

**Review by maorfr — APPROVED** [review]

**Pull request merged**